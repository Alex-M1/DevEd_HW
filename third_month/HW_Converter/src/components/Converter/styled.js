import styled from 'styled-components';

export const StConverter = styled.div`
  width:500px;
  display: grid;
  grid-template-columns: 5fr 5fr 2fr;
  justify-items: center;
  align-items: center;
  padding: 10px;
`;
