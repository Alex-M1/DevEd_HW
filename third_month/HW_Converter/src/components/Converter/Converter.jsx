import React from 'react';
import PropTypes from 'prop-types';
import { Button, ConvertItem } from '../__common';
import { StConverter } from './styled';

function Converter({
  values,
  primary,
  secondary,
  inputValue,
  resultValue,
  setPrimary,
  setSecondary,
  setInputValue,
  onButtonClick,
}) {
  return (
    <StConverter>
      <ConvertItem
        inputValue={inputValue}
        setInputValue={setInputValue}
        values={values}
        dropValue={primary}
        onChange={setPrimary}
      />
      <ConvertItem
        inputValue={resultValue}
        values={values}
        dropValue={secondary}
        onChange={setSecondary}
        readOnly
      />
      <Button onClick={onButtonClick} />
    </StConverter>
  );
}

export default Converter;

Converter.propTypes = {
  values: PropTypes.arrayOf(PropTypes.string).isRequired,
  primary: PropTypes.string.isRequired,
  secondary: PropTypes.string.isRequired,
  inputValue: PropTypes.string.isRequired,
  resultValue: PropTypes.string.isRequired,
  setPrimary: PropTypes.func.isRequired,
  setSecondary: PropTypes.func.isRequired,
  setInputValue: PropTypes.func.isRequired,
  onButtonClick: PropTypes.func.isRequired,
};
