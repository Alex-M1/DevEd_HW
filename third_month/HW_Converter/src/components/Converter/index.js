import { connect } from 'react-redux';
import { setInputValue, setPrimary, setSecondary, setConvert, getCurrency } from '../../store/actions';
import { getInputValue, getPrimaryValue, getResultValue, getSecondaryValue } from '../../store/selectors';
import Converter from './Converter';

const mapStateToProps = (state, props) => ({
  primary: getPrimaryValue(state, props.type),
  secondary: getSecondaryValue(state, props.type),
  inputValue: getInputValue(state, props.type),
  resultValue: getResultValue(state, props.type),
});

const mapDispatchToProps = (dispatch, props) => ({
  setPrimary: (value) => dispatch(setPrimary(props.type, value)),
  setSecondary: (value) => dispatch(setSecondary(props.type, value)),
  setInputValue: (value) => dispatch(setInputValue(props.type, value)),
  onButtonClick: () => {
    if (props.type === 'length') {
      dispatch(setConvert(props.type));
    } else if (props.type === 'currency') {
      dispatch(getCurrency());
    }
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(Converter);
