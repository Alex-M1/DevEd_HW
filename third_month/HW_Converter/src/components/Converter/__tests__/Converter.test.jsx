import { mount, shallow } from 'enzyme';
import React from 'react';
import Converter from '../Converter';

describe('Form', () => {
  const props = {
    values: ['UAH', 'EUR', 'USD', 'RUB'],
    primary: '',
    secondary: '',
    inputValue: '',
    resultValue: '',
    setPrimary: jest.fn(),
    setSecondary: jest.fn(),
    setInputValue: jest.fn(),
    onButtonClick: jest.fn(),
  };
  afterEach(() => {
    props.setPrimary.mockClear();
    props.setSecondary.mockClear();
    props.setInputValue.mockClear();
    props.onButtonClick.mockClear();
  });
  it('Should match snapshot', () => {
    const component = shallow(<Converter {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StConverter', () => {
    const component = mount(<Converter {...props} />);
    expect(component.find('styled__StConverter')).toHaveLength(1);
  });
  it('Should have Input', () => {
    const component = mount(<Converter {...props} />);
    expect(component.find('ConvertItem')).toHaveLength(2);
  });
  it('Should have Dropdown', () => {
    const component = mount(<Converter {...props} />);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('Should have call setPrimary', () => {
    const component = mount(<Converter {...props} />);
    component.props().setPrimary();
    expect(component.props().setPrimary).toHaveBeenCalledTimes(1);
  });
  it('Should have call setSecondary', () => {
    const component = mount(<Converter {...props} />);
    component.props().setSecondary();
    expect(component.props().setSecondary).toHaveBeenCalledTimes(1);
  });
  it('Should have call setInputValue', () => {
    const component = mount(<Converter {...props} />);
    component.props().setInputValue();
    expect(component.props().setInputValue).toHaveBeenCalledTimes(1);
  });
  it('Should have call onButtonClick', () => {
    const component = mount(<Converter {...props} />);
    component.props().onButtonClick();
    expect(component.props().onButtonClick).toHaveBeenCalledTimes(1);
  });
});
