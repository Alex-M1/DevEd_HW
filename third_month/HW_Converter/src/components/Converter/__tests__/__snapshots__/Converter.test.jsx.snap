// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `"<div class=\\"styled__StConverter-sc-gbnl2c-0 hjwyPX\\"><div class=\\"styled__StConvertItem-sc-cq8jb4-0 calHuE\\"><input value=\\"\\" class=\\"styled__StInput-sc-c3z9ep-0 feLlzU\\"/><select class=\\"styled__StSelect-sc-7l3rob-0\\"><option value=\\"UAH\\" class=\\"styled__StOption-sc-7l3rob-1\\">UAH</option><option value=\\"EUR\\" class=\\"styled__StOption-sc-7l3rob-1\\">EUR</option><option value=\\"USD\\" class=\\"styled__StOption-sc-7l3rob-1\\">USD</option><option value=\\"RUB\\" class=\\"styled__StOption-sc-7l3rob-1\\">RUB</option></select></div><div class=\\"styled__StConvertItem-sc-cq8jb4-0 calHuE\\"><input value=\\"\\" readonly=\\"\\" class=\\"styled__StInput-sc-c3z9ep-0 feLlzU\\"/><select class=\\"styled__StSelect-sc-7l3rob-0\\"><option value=\\"UAH\\" class=\\"styled__StOption-sc-7l3rob-1\\">UAH</option><option value=\\"EUR\\" class=\\"styled__StOption-sc-7l3rob-1\\">EUR</option><option value=\\"USD\\" class=\\"styled__StOption-sc-7l3rob-1\\">USD</option><option value=\\"RUB\\" class=\\"styled__StOption-sc-7l3rob-1\\">RUB</option></select></div><button class=\\"styled__StButton-sc-z42suv-0 dyLAGI\\">Convert</button></div>"`;
